### PR TITLE
Remove the redundant shift during the loss computation in the Moshi m…

### PR DESCRIPTION
**Pull request by glynpu**
# What does this PR do?

Correct the loss computation process in the Moshi model to apply the shift only once, as it is currently being applied twice.

Because the class name **MoshiForCausalLM** contains '**ForCausalLM**', according to the mapping rules in **LOSS_MAPPING**, the **self.loss_function** used in the forward function of **MoshiForCausalLM** should be **ForCausalLMLoss**. As a result, logits and labels are shifted **twice**: once before calling self.loss_function and once inside self.loss_function. This leads to **tokens < n - 1 predicting n** instead of the expected behavior where **tokens < n predict n**.

This PR removes the shift before the self.loss_function call.

**References:**
**LOSS_MAPPING**:
https://github.com/huggingface/transformers/blob/9e125d9a2e696967b4d401dbfea24a4d40058017/src/transformers/loss/loss_utils.py#L130-L135

ForCausalLMLoss:
https://github.com/huggingface/transformers/blob/9e125d9a2e696967b4d401dbfea24a4d40058017/src/transformers/loss/loss_utils.py#L33-L57

### CODE REVIEW

**Comment by Rocketknight1** [comment]
cc @eustlb!

**Comment by eustlb** [comment]
Hey @glynpu, thank you so much for your patience. 
I am allocating time this week to a bug fixing sprint on Moshi, I'll review this PR at the same time 🤗
